packages/typescript/src/lib/Typescript.ts

```diff
@@ -123,30 +123,54 @@ export class FederatedTypesPlugin {
 
   private extractTypes() {
     if (this.exposedComponents) {
-      const normalizedFileNames = Object.values(this.exposedComponents)
-        .map((exposed) => {
-          const [rootDir, entry] = exposed.split(/\/(?=[^/]+$)/);
-
+      const exposeSrcToDestMap: Record<string, string> = {};
+      const normalizedFileNames = Object.entries(this.exposedComponents)
+        // './Component': 'path/to/component' -> ['./Component', 'path/to/component']
+        .map(([exposeDest, exposeSrc]) => {
+          const [rootDir, entry] = exposeSrc.split(/\/(?=[^/]+$)/);
           const cwd = this.webpackCompilerOptions.context || process.cwd();
           const normalizedRootDir = path.resolve(cwd, rootDir);
           const filenameWithExt = this.resolveFilenameWithExtension(
             normalizedRootDir,
             entry
           );
-
-          return path.resolve(cwd, rootDir, filenameWithExt);
+          const pathWithExt = path.resolve(cwd, rootDir, filenameWithExt);
+          exposeSrcToDestMap[pathWithExt] = exposeDest;
+          return pathWithExt;
         })
         .filter((entry) => /\.tsx?$/.test(entry));
 
       const host = ts.createCompilerHost(this.tsCompilerOptions);
       const originalWriteFileFn = host.writeFile;
 
-      host.writeFile = (...args) => {
-        const [filename, data] = args;
+      host.writeFile = (
+        filename,
+        text,
+        writeOrderByteMark,
+        onError,
+        sourceFiles,
+        data
+      ) => {
+        // for exposes: { "./expose/path": "path/to/file" }
+        // force typescript to write compiled output to "@mf-typescript/expose/path"
+        const newFileName = `${
+          exposeSrcToDestMap[sourceFiles?.[0].fileName || '']
```

Comment:
I'm not completely sure but I think `sourceFiles?.[0].fileName` is just the name of the file instead of absolute path which is what got stored in `exposeSrcToDestMap` from `line#138`.

If that is the case, newFileName always resolves to `undefined.d.ts`

Reply:
no the fileName is the full path. Equally I could use `path`
```shell
[
  SourceFileObject2 {
    ...
    fileName: '/Users/haydonlam/Documents/projects/home-fe-test/test/counter.ts',
    path: '/users/haydonlam/documents/projects/home-fe-test/test/counter.ts',
    resolvedPath: '/users/haydonlam/documents/projects/home-fe-test/test/counter.ts',
    originalFileName: '/Users/haydonlam/Documents/projects/home-fe-test/test/counter.ts',
    original: SourceFileObject2 {
      fileName: '/Users/haydonlam/Documents/projects/home-fe-test/test/counter.ts',
      path: '/users/haydonlam/documents/projects/home-fe-test/test/counter.ts',
      resolvedPath: '/users/haydonlam/documents/projects/home-fe-test/test/counter.ts',
      originalFileName: '/Users/haydonlam/Documents/projects/home-fe-test/test/counter.ts',
  }
]
```